bin/run.sh

```diff
@@ -77,15 +79,54 @@ run_tests() {
     # test scripts may be (old) xxxx_test.sh
     [[ -f "$test_file" ]] || test_file="${slug//-/_}_test.sh"
 
-    sed -i 's/load bats-extra.bash/load bats-extra/' "$test_file" || true
+    perl -i -pe 's/(load bats-extra)\.bash/$1/' "$test_file"
 
     echo "Test output:"
 
     bats --tap "$test_file" 2>&1 | tee "$output_file" || true
 
+    get_test_bodies "$test_file"
+
     echo "Test run ended. Output saved in $output_file"
 
-    cd ~-
+    cd -
+}
+
+get_test_bodies() {
+    local test_file=$1
+    local name line
+    local state="out"
+    local body=() 
+    test_bodies=()
+
+    local start_test_re='^@test ['\''"](.+)['\''"] \{[[:blank:]]*$'
```

Comment:
Is this JSON parsing? Can we use `jq` or `awk` for parsing?

Reply:
This is parsing the bats test file.

Reply:
If you want to parse in `awk` then read null-delineated data...
```
while IFS='' read -d '' name; read -d '' body; do printf '[%s][%s]\n' "$name" "$body"; done < <(
    awk '
        $0 ~ /^}$/ {i=0; printf "\00"}
        i {print}
        $1 == "@test" && $(NF) == "{" {
            sub(/^@test "/, "")
            sub(/" {$/, "")
            print $0, "\00"
            i=1
    } ' file.bats
)
```

Not sure if you find that harder or easier to read.

Reply:
I think I'm satisfied parsing the file in bash. I think the code is pretty clear.